clients/client-python/gravitino/client/gravitino_client.py

```diff
@@ -99,3 +102,112 @@ def enable_catalog(self, name: str):
+        """Retrieves a job template by its name.
+
+        Args:
+            job_template_name: The name of the job template to retrieve.
+
+        Returns:
+            The JobTemplate object corresponding to the specified name.
+
+        Raises:
+            NoSuchJobTemplateException: If no job template with the specified name exists.
+        """
+        return self.get_metalake().get_job_template(job_template_name)
+
+    def delete_job_template(self, job_template_name: str) -> bool:
+        """
+        Deletes a job template by its name. This will remove the job template from Gravitino, and it
+        will no longer be available for execution. Only when all the jobs associated with this job
+        template are completed, failed or cancelled, the job template can be deleted successfully,
+        otherwise it will return false.
+
+        The deletion of a job template will also delete all the jobs associated with this template.
+
+        Args:
+            job_template_name: The name of the job template to delete.
+
+        Returns:
+            bool: True if the job template was deleted successfully, False if the job template
+            does not exist.
+
+        Raises:
+            InUseException: If there are still queued or started jobs associated with this job template.
+        """
+        return self.get_metalake().delete_job_template(job_template_name)
+
+    def list_jobs(self, job_template_name: str = None) -> List[JobHandle]:
+        """Lists all the jobs in the current metalake.
+
+        Args:
+            job_template_name: Optional; if provided, filters the jobs by the specified job template name.
+
+        Returns:
+            A list of JobHandle objects representing the jobs in the metalake.
+        """
+        return self.get_metalake().list_jobs(job_template_name)
+
+    def get_job(self, job_id: str) -> JobHandle:
+        """Retrieves a job by its ID.
+
+        Args:
+            job_id: The ID of the job to retrieve.
+
+        Returns:
+            The JobHandle object corresponding to the specified job ID.
+
+        Raises:
+            NoSuchJobException: If no job with the specified ID exists.
+        """
+        return self.get_metalake().get_job(job_id)
+
+    def run_job(self, job_template_name: str, job_conf: Dict[str, str]) -> JobHandle:
+        """Runs a job using the specified job template and configuration.
+
+        Args:
+            job_template_name: The name of the job template to use for running the job.
+            job_conf: A dictionary containing the configuration for the job.
+
+        Returns:
+            A JobHandle object representing the started job.
+
+        Raises:
+            NoSuchJobTemplateException: If no job template with the specified name exists.
+        """
+        return self.get_metalake().run_job(job_template_name, job_conf)
+
+    def cancel_job(self, job_id: str) -> JobHandle:
+        """Cancels a running job by its ID.
```

Comment:
What if the status of a job is `queue`?

Reply:
It will be cancelled if possible, it relies on the implemented `JobExecutor` whether it can support canceling queued jobs.